reporter/src/main/kotlin/reporters/AbstractNoticeReporter.kt

```diff
@@ -111,6 +89,43 @@ abstract class AbstractNoticeReporter : Reporter() {
         return result.toSortedMap()
     }
 
+    private fun writeNoticeReport(outputFile: File, noticeReport: AbstractNoticeReporter.NoticeReport) {
```

Comment:
Can't you drop the `AbstractNoticeReporter.` prefix here, as it's defined in the same class?

Should we swap arguments, to match the order in which variables are used?